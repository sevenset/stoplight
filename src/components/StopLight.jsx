import React, { useState } from "react";
import StopLightCSS from "./StopLight.module.css";
import LightCSS from "./Light.module.css";

const StopLight = () => {
  const [currColor, setCurrColor] = useState("red");

  setTimeout(updateLight, 3000);

  function updateLight() {
    if (currColor === "red") {
      setCurrColor("green");
    } else if (currColor === "green") {
      setCurrColor("yellow");
    } else if (currColor === "yellow") {
      setCurrColor("red");
    }
  }

  return (
    <div className={StopLightCSS.background}>
      <div className={StopLightCSS.light_box}>
        <Light
          onClassName={LightCSS.red_light_active}
          offClassName={LightCSS.red_light}
          isOn={currColor === "red"}
        />
        <Light
          onClassName={LightCSS.yellow_light_active}
          offClassName={LightCSS.yellow_light}
          isOn={currColor === "yellow"}
        />
        <Light
          onClassName={LightCSS.green_light_active}
          offClassName={LightCSS.green_light}
          isOn={currColor === "green"}
        />
      </div>
    </div>
  );
};

const Light = ({ isOn, onClassName, offClassName }) => {
  return (
    <div className={LightCSS.light_box}>
      <div className={isOn ? onClassName : offClassName}></div>
      <div className={LightCSS.light_top}></div>
    </div>
  );
};

export default StopLight;
