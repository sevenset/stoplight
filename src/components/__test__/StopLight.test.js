import React from "react";
import ReactDOM from "react-dom";
import StopLight from "../StopLight";

import { cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders StopLight without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<StopLight></StopLight>, div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<StopLight></StopLight>).toJSON();
  expect(tree).toMatchSnapshot();
});
