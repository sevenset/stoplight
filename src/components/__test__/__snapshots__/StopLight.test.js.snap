// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="background"
>
  <div
    className="light_box"
  >
    <div
      className="light_box"
    >
      <div
        className="red_light_active"
      />
      <div
        className="light_top"
      />
    </div>
    <div
      className="light_box"
    >
      <div
        className="yellow_light"
      />
      <div
        className="light_top"
      />
    </div>
    <div
      className="light_box"
    >
      <div
        className="green_light"
      />
      <div
        className="light_top"
      />
    </div>
  </div>
</div>
`;
